### PR TITLE
Add transformer for edx.course.enrollment.activated event

### PR DESCRIPTION
**Pull request by mumarkhan999**
**Trello Link:** [https://trello.com/c/zHVxuG6e/21-enrollment-events-edxcourseenrollmentactivated]()

**Description:** 
Includes a transformer for edx_course_enrollment_activated event.

**Checks before merge:**

- [x] Reviewed
- [x] Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/transformers/course_transformers.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""
```
Comment (tasawernawaz) [inline openedx/features/caliper_tracking/transformers/course_transformers.py]: rename this file `enrollment_transformers.py`

File: openedx/features/caliper_tracking/transformers/course_transformers.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Transformers for all course related events.
+"""
+
+
+def edx_course_enrollment_activated(current_event, caliper_event):
+    caliper_event['ip'] = current_event['ip']
+    if 'ip' in caliper_event['actor']:
+        del caliper_event['actor']['ip']
```
Comment (tasawernawaz) [inline openedx/features/caliper_tracking/transformers/course_transformers.py]: already removed from base transformer, please remove this check

Comment (asamolion) [inline openedx/features/caliper_tracking/transformers/course_transformers.py]: Why is the deletion required? If the `ip` key is not required in `caliper_event` then it should not be there.

File: openedx/features/caliper_tracking/transformers/course_transformers.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    caliper_event['actor']['type'] = 'Person'
```
Comment (asamolion) [inline openedx/features/caliper_tracking/transformers/course_transformers.py]: You can group some of these into a `dict.update()` call.

File: openedx/features/caliper_tracking/transformers/course_transformers.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    caliper_event['type'] = 'Activated'
+    caliper_event['object'] = current_event['event']
+    caliper_event['status'] = current_event['event']['mode']
+    caliper_event['page'] = current_event['page']
+    return caliper_event
```
Comment (tasawernawaz) [inline openedx/features/caliper_tracking/transformers/course_transformers.py]: simplify this 
```
caliper_event = {
    'key' : 'value',
    ...
}
```

File: openedx/features/caliper_tracking/transformers/course_transformers.py (hunk 2 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/course_transformers.py]: It has been removed
As in the early version of master
base_transformer ip has been sent in actor's 
dict but now it's not

File: openedx/features/caliper_tracking/transformers/course_transformers.py (hunk 1 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/course_transformers.py]: done in the latest commit

File: openedx/features/caliper_tracking/transformers/course_transformers.py (hunk 2 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/course_transformers.py]: ok

File: openedx/features/caliper_tracking/transformers/course_transformers.py (hunk 3 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/course_transformers.py]: done in the latest commit

File: openedx/features/caliper_tracking/caliper_config.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
 from openedx.features.caliper_tracking.transformers.bookmark_transformers import edx_bookmark_listed
-
```
Comment (tasawernawaz) [inline openedx/features/caliper_tracking/caliper_config.py]: nit: remove extra empty line

Comment (mumarkhan999) [reply openedx/features/caliper_tracking/caliper_config.py]: It has been already removed in the latest commit.

File: openedx/features/caliper_tracking/transformers/enrollment_transformers.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+"""
+Transformers for all course related events.
+"""
+
+
+def edx_course_enrollment_activated(current_event, caliper_event):
+    caliper_event.update({
+        'ip': current_event['ip'],
+        'object': current_event['event'],
```
Comment (arslanhashmi) [inline openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: @tasawernawaz @mumarkhan999 don't we need `object` as a dictionary?

Comment (tasawernawaz) [reply openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: I assume `current_event['event']` will return dictionary.

**Comment by tasawernawaz** [comment]
@asamolion I think it would be better if you had approved it before merging, also I think owner should take responsibility of merging the branch.

P.S please delete your branch from github after merging.

**Comment by asamolion** [comment]
@tasawernawaz You had already approved it so I merged. We can go with developer responsibility for merging.

**Comment by asamolion** [comment]
@mumarkhan999 
I checked the output of your code

Here is the output of your code
```
{
  "@context": "http://purl.imsglobal.org/ctx/caliper/v1p1",
  "action": "Activated",
  "actor": {
    "id": "http://localhost:8000/u/honor",
    "type": "Person"
  },
  "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36",
  "edx_event_type": "edx.course.enrollment.activated",
  "eventTime": "2018-10-31T13:44:19.111131",
  "host": "747befdb3d9d",
  "id": "urn:uuid:7d98601f-ca50-4c85-bba4-efa09eebb3a3",
  "ip": "172.18.0.1",
  "object": {
    "course_id": "course-v1:edx+cs-101+2018",
    "mode": "audit",
    "user_id": 6
  },
  "org_id": "edx",
  "page": null,
  "path": "/change_enrollment",
  "referer": "http://localhost:18000/courses/course-v1:edx+cs-101+2018/about",
  "session": "8205ce7ea40a0619955e15f072dc33e2",
  "status": "audit",
  "type": "Membership",
  "user_id": 6
}
```

while this is the expected output on the Trello board
```
{
  "@context": "http://purl.imsglobal.org/ctx/caliper/v1p1",
  "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36",
  "course_id": "course-v1:edx+cs-101+2018",
  "dateCreated": "2018-11-01T06:00:00.000Z",
  "edx_event_type": "edx.course.enrollment.activated",
  "host": "747befdb3d9d",
  "id": "https://example.edu/terms/201801/courses/7/sections/1/rosters/1/members/554433",
  "ip": "172.18.0.1",
  "member": {
    "id": "https://example.edu/users/554433",
    "type": "Person"
  },
  "org_id": "edx",
  "organization": {
    "id": "https://example.edu/terms/201801/courses/7/sections/1",
    "type": "CourseSection"
  },
  "page": null,
  "path": "/change_enrollment",
  "referer": "http://localhost:18000/courses/course-v1:edx+cs-101+2018/about",
  "roles": [
    "Learner"
  ],
  "session": "8205ce7ea40a0619955e15f072dc33e2",
  "status": "Audit",
  "type": "Membership",
  "user_id": 6
}
```
There are a few keys missing and different between the two.
What is your take on this?

**Comment by mumarkhan999** [comment]
@asamolion 
for the time being it has been decided that no extra fields will be added in the calliper event as the missing fields are also mentioned on the trello board [link](https://trello.com/c/zHVxuG6e/21-enrollment-events-edxcourseenrollmentactivated)
. Only those fields will be transformed which are available in current_event

**Comment by asamolion** [comment]
@mumarkhan999 Alright. Got it.

**Pull request merged**